### ISSUE TITLE
TFS2017 - install fails

### ISSUE THREAD

**Issue by Pitterling** [GitTools/GitVersion#1096]
trying to install on upgraded TFS fails with error messages:

Task definition with id e5983830-3f75-11e5-82ed-81492570a08e is a built-in task definition and cannot be replaced.

extention was installed on TFS2015, after installation of TFS2017 it is recommended to remove and reinstall all extentions. Now i'm facing this error message. Any clue?

**Comment by pascalberger** [comment]
@Pitterling You deinstalled the extension and later reinstalled it from marketplace? Or did you use the tfx-cli tool?

**Comment by Pitterling** [comment]
@pascalberger initially i installed it by tfx-cli ... later on i upgraded by Marketplace successfully. This was all done on TFS2015. Now i upgraded to TFS2017 .. in the issue list it is recommended to reinstall all extensions. That's what i tried and failed finally. Uninstall went trough .. now no new installation is possible because of this error. The gitversion extension is not in the list of installed extensions.

Is the listed id something you specified?

**Comment by pascalberger** [comment]
@Pitterling Yes, this is the ID of the GitVersion task. See https://github.com/GitTools/GitVersion/blob/master/src/GitVersionTfsTask/task.json#L2.

Do you have this issue only with GitVersion or also with other extensions providing build tasks?

**Comment by Pitterling** [comment]
@pascalberger i have only 2 non-MS extensions .. The other one deinstalled/installed without any issues.

**Comment by pascalberger** [comment]
I've checked with Microsoft and it seems like we have an ID conflict between GitVersion and one of the tasks Microsoft ships. I'll check how we can resolve this without breaking too much for existing users.

**Comment by bryanmacfarlane** [comment]
@Pitterling - Hello, bryanmac from microsoft here.  We're trying to find the conflict and I can't seem to locate the conflicting id.  Can you send us an email so we can diagnose further?  send an email to bryanmac and that's at microsoft dot com

**Comment by pascalberger** [comment]
This issue was caused by the task being first installed through tfx-cli.

**Comment by DanielTheCoder** [comment]
Hi @pascalberger 
how have you fixed the problem?
We are in a similar situation
We are using TFS 2017.3.1.
Before we upgraded to TFS2017.3.1, we've installed GitVersion using tfx-cli.
After the upgrade we have tried to upgrade the extension via TFS Gallery on the collection which not worked because of the same error. Because of the error we've uninstalled the extension via gallery from the collection and updated the extension to the latests version on tfs self.

Now we cannot install GitVersion to the collection because of the same error message.

We don't see GitVersion as installed extension on the collection.
We see GitVersion as exisiting extension using tfx-cli
Trying to delete the existing extension using tfx-cli does not work, even there is an issue on tfx-cli that delete is not working.
